stader/node/manage-fee-recipient.go

```diff
@@ -119,6 +121,33 @@ func (m *manageFeeRecipient) run() error {
 		return err
 	}
 
+	currentBlock, err := eth1.GetCurrentBlockNumber(m.c)
+	if err != nil {
+		return fmt.Errorf("error GetCurrentBlockNumber: %w", err)
+	}
+
+	pnr, err := services.GetPermissionlessNodeRegistry(m.c)
+	if err != nil {
+		return fmt.Errorf("error GetPermissionlessNodeRegistry: %w", err)
+	}
+
+	operatorID, err := node.GetOperatorId(pnr, nodeAccount.Address, nil)
+	if err != nil {
+		return fmt.Errorf("error GetOperatorId: %w", err)
+	}
+
+	lastChangeBlock, err := node.GetSocializingPoolStateChangeBlock(pnr, operatorID, nil)
+	if err != nil {
+		return fmt.Errorf("error GetSocializingPoolStateChangeBlock: %w", err)
+	}
+
+	nextUpdatableBlock := lastChangeBlock.Add(lastChangeBlock, blockPerThreeEpoch).Uint64()
+
+	if currentBlock < nextUpdatableBlock {
```

Comment:
With this code, you will be skipping the fee recipient check completely for 3 epochs. Rather change the logic to
```
if feeRecipientInfo.IsInSocializingPool {
    if(currentBlock > nextUpdatableBlock){
      correctFeeRecipient = feeRecipientInfo.SocializingPoolAddress
    }
    else {
      correctFeeRecipient = feeRecipientInfo.FeeDistributorAddress
    }
  } else {
    if(currentBlock > nextUpdatableBlock){
      correctFeeRecipient = feeRecipientInfo.FeeDistributorAddress
      
    }
    else {
      correctFeeRecipient = feeRecipientInfo.SocializingPoolAddress
    }
```
This will check and delay the change for opt-in opt-out